File: Python/Prefix Sum/Number of Ways to Split Array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        prefSum=[]
        prefSum.append(nums[0])
        for i in range(1, len(nums)):
            prefSum.append(prefSum[i-1]+nums[i])
        cnt =0
        n =len(nums)
        for i in range(n-1):
            leftSum=prefSum[i]
            rightSum=prefSum[n-1]-prefSum[i]
            if leftSum>=rightSum:
                cnt+=1
        return cnt
            
        

File: CPP/Prefix Sum Array/Number of Ways to Split Array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        vector<long long>prefSum;
        prefSum.push_back(nums[0]);
        for(int i=1;i<nums.size();i++){
            prefSum.push_back(prefSum[i-1]+nums[i]);
        }
        int cnt =0;
        int n =nums.size();
        for(int i=0;i<nums.size()-1;i++){
            long long leftSum=prefSum[i];
            long long rightSum=prefSum[n-1]-prefSum[i];
            if(leftSum>=rightSum)cnt++;
        }
        return cnt;
    }
};
